### PR TITLE
hale schema definition as Json

### PR DESCRIPTION
**Pull request by stempler**

### CODE REVIEW

File: common/plugins/eu.esdihumboldt.hale.common.core/src/eu/esdihumboldt/hale/common/core/io/impl/SimpleValue.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2013 Data Harmonisation Panel
```
Comment (florianesser) [inline common/plugins/eu.esdihumboldt.hale.common.core/src/eu/esdihumboldt/hale/common/core/io/impl/SimpleValue.java]: Outdated header?

File: common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaReaderJson.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+				schema = it.next();
+				reporter.setSuccess(true);
+			}
+			else {
+				reporter.setSuccess(false);
+				reporter.setSummary("No schema definition found");
+			}
+
+			if (it.hasNext()) {
+				// FIXME what about multiple schemas???
```
Comment (florianesser) [inline common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaReaderJson.java]: `SchemaToXml`/`XmlToSchema` use different root tags to differentiate between one/multiple schemas in a single file. Could something similar be done here?

File: common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaWriterJson.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		try (OutputStream out = getTarget().getOutput();
+				Writer writer = new OutputStreamWriter(out, StandardCharsets.UTF_8)) {
+			// create DOM
+			JsonStreamBuilder builder = new JsonStreamBuilder(writer, true);
+			new SchemaToJson().schemasToJson(builder, getSchemas().getSchemas(), null);
+
+			reporter.setSuccess(true);
+		} catch (Exception e) {
+			// XXXXXXXXXXXXXXXXXX
+			e.printStackTrace();
```
Comment (florianesser) [inline common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaWriterJson.java]: might be removed?

File: common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaReaderJson.java (hunk 2 above)
Comment (florianesser) [inline common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaReaderJson.java]: Oh, looking further down leads me to believe that a similar approach is already taken by `SchemaToJson`/`JsonToSchema`...

Comment (stempler) [inline common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaReaderJson.java]: Yes, the actual problem is that a schema reader produces one Schema and not multiple. That might be something that needs to be solved on a different level.

File: common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaWriterJson.java (hunk 3 above)
Comment (stempler) [inline common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/json/HaleSchemaWriterJson.java]: A yes, left the Xes in there so I would not forget :wink:

**Pull request merged**